[Misc] Update guided decoding logs to debug

sfbemerk opened the pull request:
The json schema as part of guided decoding details can contain sensitive user inputs. I believe user inputs should not be logged when the engine is started with `disable_log_requests=True`.

These logs have existed for a while at DEBUG level but only made visible at INFO level with v0.8.0 (#12955). I would be fine with reverting the relevant line's log level back to DEBUG instead of my proposed change.

robertgshaw2-redhat commented on vllm/engine/async_llm_engine.py:
```diff
@@ -531,7 +537,7 @@ async def check_health_async(self) -> None:
         sampling_params: SamplingParams, tokenizer: AnyTokenizer,
         default_guided_backend: str, reasoning_backend: Optional[str],
-        model_config: ModelConfig) -> SamplingParams:
+        model_config: ModelConfig, *, verbose: bool) -> SamplingParams:
```
can you remove the `*`? This makes it very hard to maintain since we cannot keep track of the function signature

robertgshaw2-redhat commented:
Should we just revert this to debug? Would rather not make these functions have more options if we can avoid it

aarnphm commented:
+1 with rob, let's just make it `logger.debug`

Commit:
Log guided decoding details on DEBUG, not INFO.

Guided decoding json schema contains potentially sensitive user inputs. Do not log user inputs on INFO while ignoring `disable_log_requests=True`.

Signed-off-by: Benjamin Merkel <benjamin.merkel@tngtech.com>

aarnphm reviewed (approved):
Can you also update the title to "[Misc] Update guided decoding logs to debug"

Thanks

hmellor reviewed (approved):
LGTM

sfbemerk commented:
Thank you all!
I think the CI is failing for unrelated reasons. @hmellor could you please force-merge?
